File: mcst/While.m
classdef While < Segment
    properties
        head Segment
        body Segment
        end_ Segment
    end
    
    methods
        function self = While(head,body,end_)
            self.head = head;
            self.body = body;
            self.end_ = end_;
        end
    end
end

File: test_m/py/nodes/While.py
from mruntime import *
from test_m.py.nodes.Segment import Segment
class While(Segment):
    def __init__(self, *nargin): # retval: self
        [head, body, end_] = nargin
        nargin = len(nargin)
        self.head = head
        self.body = body
        self.end_ = end_
    @staticmethod
    def empty():
        return []
